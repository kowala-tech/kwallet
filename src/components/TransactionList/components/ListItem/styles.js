const styles = theme => ({
	avatar: {
		backgroundColor: theme.palette.primary.main,
	},
	amount: {
		textAlign: "right"
	}
});

export default styles;
