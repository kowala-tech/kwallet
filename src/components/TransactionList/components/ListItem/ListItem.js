import React from "react";
import PropTypes from "prop-types";
import { FormattedNumber } from "react-intl";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
// Icons
import Avatar from "@material-ui/core/Avatar";
import PersonIcon from "@material-ui/icons/Person";
// Component Related Imports
import styles from "./styles";
import { weiToKusd } from "../../../../transforms/currency";
import { unixToHumanTime } from "../../../../transforms/time";

const TransactionListItem = ({
	classes,
	transaction,
	toggleTransactionDetailsDialogFn
}) => {
	return (
		<ListItem button
			onClick={toggleTransactionDetailsDialogFn}>
			<ListItemAvatar>
				<Avatar className={classes.avatar}>
					<PersonIcon />
				</Avatar>
			</ListItemAvatar>
			<ListItemText
				className={classes.truncate}
				primary={transaction.txid.substring(0,16)}
				secondary={unixToHumanTime(transaction.date)}
			/>
			<ListItemText
				className={classes.amount}
				primary={
					<FormattedNumber
						minimumFractionDigits={4}
						maximumFractionDigits={18}
						value={weiToKusd(transaction.nativeAmount)}
					/>
				}
				secondary="kUSD"
			/>
		</ListItem>
	);
};

TransactionListItem.propTypes = {
	classes: PropTypes.object.isRequired,
	transaction: PropTypes.object,
	toggleTransactionDetailsDialogFn: PropTypes.func,
};

export default withStyles(styles)(TransactionListItem);
