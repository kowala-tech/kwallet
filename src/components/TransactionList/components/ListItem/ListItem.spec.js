import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import moment from "moment";
import ListItem from "./ListItem";

describe("ListItem", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<ListItem
					amount={12.8544321}
					timestamp={moment("2018-01-01T00:00:00-00:00").utc().unix()}
					index={0}
					toggleTransactionDetailsDialogFn={jest.fn()}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
