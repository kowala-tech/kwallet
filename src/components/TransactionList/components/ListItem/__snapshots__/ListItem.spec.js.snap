// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem it should match the snapshot 1`] = `
<TransactionListItem
  amount={12.8544321}
  classes={
    Object {
      "amount": "TransactionListItem-amount-2",
      "avatar": "TransactionListItem-avatar-1",
    }
  }
  index={0}
  timestamp={1514764800}
  toggleTransactionDetailsDialogFn={[MockFunction]}
/>
`;
